### DISCUSSION TITLE
[TM] Improve suggestions: Do not suggest if the translation is the same as the source

### DISCUSSION THREAD

**Discussion by emmapeel2** [WeblateOrg/weblate#10140] [Ideas]
Using the Machine translation suggestions feature (much improved now that you can see the origin of the suggestions!) in our project we still receive suggestions that are not good enough.

### Suggestion == source

For example, in the memory there are translations that are just equal to the source.

Our components have created this problem many times, because we have translations that need to have all strings, so we have the English string when we dont have translations. When importing this projects, I import the translations and do a Bulk Edit to convert all the same-check strings to 'Needs edit'. But the English string stays in the memory for the translation.

I think that the Translation Memory should not suggest strings that have no changes with the English string. If I need to have the English string as a placeholder, I can do it myself on the project with the corresponding Add-on. If it is in a translation memory, it is most probably a bad translation.

### Suggestion == translation

If a translator adds a translation that is the same as a suggestion, the suggestion should dissapear.

As a translation mantainer, I try always to recover the strings with small modifications or translated in other parts of our project to save translators of repetitive jobs. So I go through the 'strings with suggestions' list, but instead of having the Suggestions tab open, I use the 'Automatic translations' tab, so I can see the provenance of strings, how many different versions there are, difference between the source of the suggestion and the current string source, etc.

It would be nice that when I add a translation that is a suggestion, this suggestion gets removed somehow.

### Learn from rejection

If a suggestion gets rejected several times, the machine translation should flag it somehow and eventually stop using it. Same with the accepted suggestions, they should become more prominent.

**Comment by nijel** [comment]
Suggestion == source is a valid translation sometimes (there are words which are same between languages or this could be product names). When starting with a blank translation, I'd expect the translation memory to fill me even strings which do not change when translating.

Suggestion == translation should not be possible to add. What can happen though is that the translation is changed to the same string by other means (for example in repository). Then there might be a no-change suggestion. Such suggestions should disappear within a day (see https://docs.weblate.org/en/latest/user/translating.html#suggestions).

**Reply by emmapeel2** [reply]
> Suggestion == source is a valid translation sometimes (there are words which are same between languages or this could be product names). When starting with a blank translation, I'd expect the translation memory to fill me even strings which do not change when translating.

If a string is the same in translation and source, not having a translation will show it as it is in the source for many of our components, so there is no difference in the results. 

But we have other components in which we need to put the English string in the translation if there is still no translation for that language. This makes the translation memory suggest this 'placeholder translations' to other components.

In some formats, if the string is not translated to X language, the user can choose the language they want to see it translated to, and this language can be different than English.
